### PR TITLE
Guard use of OpenMP in rocblas test

### PR DESCRIPTION
**Pull request by renjithravindrankannath**
Guard use of OpenMP in rocblas test

### CODE REVIEW

File: var/spack/repos/builtin/packages/rocblas/0005-omp-for-test.patch [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
++  target_link_libraries( rocblas-test PRIVATE "${ROCM_OPENMP_EXTRAS_DIR}/lib/libomp.so" hip::host hip::device )
+ endif( )
+ 
+ if( CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+@@ -179,7 +180,7 @@ if (NOT WIN32)
+ else()
+   list( APPEND COMMON_LINK_LIBS "libomp")
+ endif()
+-target_link_libraries( rocblas-test PRIVATE ${COMMON_LINK_LIBS} )
++target_link_libraries( rocblas-test PRIVATE "${ROCM_OPENMP_EXTRAS_DIR}/lib/libomp.so" ${COMMON_LINK_LIBS} )
```
Comment (cgmb) [inline var/spack/repos/builtin/packages/rocblas/0005-omp-for-test.patch]: Isn't `OpenMP::OpenMP_CXX` already in `${COMMON_LINK_LIBS}`? [rocBLAS adds it after doing `find_package(OpenMP)`](https://github.com/ROCmSoftwarePlatform/rocBLAS/blob/rocm-5.4.3/clients/CMakeLists.txt#L76-L84). I definitely appreciate the workaround that you've found, but I'm still deeply confused about how OpenMP is supposed to be used.

**Review by cgmb — DISMISSED** [review]

**Comment by cgmb** [comment]
The tests are failing because rocm-openmp-extras is not listed as a dependency. You'll need to fix that. In any case, my approval is not so much about the details, but that I approve of using this workaround to get the tests working. This is an ugly hack and it should not be necessary, but... well, it is.

**Comment by cgmb** [comment]
To me, the burning question is: why can't the Spack-provided ROCm stack build [openmp-helloworld](https://github.com/ROCm-Developer-Tools/HIP-Examples/tree/rocm-5.4.3/openmp-helloworld) as-is?

**Review by tldahlgren — DISMISSED** [review]
LGTM

**Comment by tldahlgren** [comment]
@spackbot run pipeline

File: var/spack/repos/builtin/packages/rocblas/package.py [hunk 2]
```diff
@@ -259,6 +261,7 @@ def cmake_args(self):
             self.define("BUILD_CLIENTS_BENCHMARKS", "OFF"),
             self.define("BUILD_CLIENTS_SAMPLES", "OFF"),
             self.define("RUN_HEADER_TESTING", "OFF"),
+            self.define("ROCM_OPENMP_EXTRAS_DIR", self.spec["rocm-openmp-extras"].prefix),
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/rocblas/package.py]: If you want to use that dependency in `cmake_args()`, you'll need to make it at least a `build` (not just `test`) dependency.

Comment (renjithravindrankannath) [reply var/spack/repos/builtin/packages/rocblas/package.py]: Moved ROCM_OPENMP_EXTRAS_DIR to run_tests since rocm-openmp-extras is added only for test.

Comment (renjithravindrankannath) [reply var/spack/repos/builtin/packages/rocblas/package.py]: Restricted ROCM_OPENMP_EXTRAS_DIR to test

**Comment by cgmb** [comment]
I've been packaging these libraries for Debian, where rocm-openmp-extras is not available yet, so I've submitted patches to rocFFT and rocBLAS to make the dependency on OpenMP optional.

If you prefer, this is the patch to allow rocBLAS to build without OpenMP: [0001-Guard-use-of-OpenMP-to-make-it-optional.patch](https://gist.github.com/cgmb/971a24d2c8df70aefc6d35efd6f7c582). It was merged as https://github.com/ROCmSoftwarePlatform/rocBLAS/commit/45bef63fc4988d11f25c9cc84b12513c7b8abc81 and is expected in rocBLAS for ROCm 5.7.

The change for rocFFT is already committed https://github.com/ROCmSoftwarePlatform/rocFFT/commit/e7b1fe244ab0623e900b4efe75be29545df1163b and expected for ROCm 5.6.

**Comment by cgmb** [comment]
I have no strong opinions about whether to get OpenMP working via these changes or to just make rocblas-test work without OpenMP. I'm happy with whatever you prefer.

File: var/spack/repos/builtin/packages/rocblas/0005-omp-for-test.patch (hunk 1 above)
Comment (renjithravindrankannath) [reply var/spack/repos/builtin/packages/rocblas/0005-omp-for-test.patch]: @cgmb, Created a patch similar to [0001-Guard-use-of-OpenMP-to-make-it-optional.patch](https://gist.github.com/cgmb/971a24d2c8df70aefc6d35efd6f7c582).  to make openmp optional for the test.
Thanks

**Review by cgmb — APPROVED** [review]

**Review by srekolam — APPROVED** [review]

**Comment by renjithravindrankannath** [comment]
Hi @tldahlgren , Is it good to go ahead and merge.

**Review by tldahlgren — APPROVED** [review]

**Comment by renjithravindrankannath** [comment]
Thanks @tldahlgren

**Pull request merged**